implementations/rust/ockam/ockam_macros/src/lib.rs

```diff
@@ -20,25 +21,33 @@ mod node_attribute;
 mod node_test_attribute;
 mod vault_test_attribute;
 
-/// Custom derive for the `ockam_core::AsyncTryClone` trait.
+/// Implements the [`AsyncTryClone`](https://docs.rs/ockam_core/latest/ockam_core/traits/trait.AsyncTryClone.html) trait for a type.
+///
+/// The macro supports the following attributes:
+///
+/// - `#[async_try_clone(crate = "...")]`: specify a path to the crate that
+///   will be used to import the items required by the macro. This can be
+///   helpful when using the macro from an internal `ockam` crate. Defaults
+///   to `ockam`.
 ///
 /// Example of use:
 ///
 /// ```ignore
 /// #[derive(ockam::AsyncTryClone)]
+/// #[async_try_clone(crate = "ockam")]
 /// pub struct MyStruct {
 ///     a: u32,
 /// }
 /// ```
-#[proc_macro_derive(AsyncTryClone)]
+#[proc_macro_derive(AsyncTryClone, attributes(async_try_clone))]
 pub fn async_try_clone_derive(input: TokenStream) -> TokenStream {
     let input = parse_macro_input!(input as DeriveInput);
     async_try_clone_derive::expand(input)
         .unwrap_or_else(to_compile_errors)
         .into()
 }
 
-/// Implements Message trait for a type.
+/// Implements the [`Message`](https://docs.rs/ockam_core/latest/ockam_core/trait.Message.html) trait for a type.
```

Comment:
Offtopic: I wonder if Message could/should automatically derive serde::{Serialize, Deserialize} traits?

Reply:
What I noticed, is when I need to create Message struct, I need to add Serialize, Deserialize traits to this struct derivation list + often I need to add serde as a Cargo dependency, and often it's the only reason I should do that
@adrianbenavides @thomcc thoughts?

Reply:
That would make sense to me, although I don't know if doing that would interfere with the serde derive. I can do some tests and see how it goes.

Reply:
> What I noticed, is when I need to create Message struct, I need to add Serialize, Deserialize traits to this struct derivation list + often this leads to adding serde as a Cargo dependency (dependency that otherwise wouldn't be there) @adrianbenavides @thomcc thoughts?

I would say you can't avoid adding that dependency since you need to have it available in the crate you are using the macro.

Reply:
I remember we had some discussion towards reexporting dependencies. Did we decide not to do that? @thomcc

Reply:
Yeah, re-exporting deps isn't good. If we want to auto-impl derive, I think a better way would be a blanket impl.

But doing it separately is more flexible, since it's not rare to need to manually write serialize/deserialize implementations.